shared/js/moz_intl.js

```diff
@@ -187,36 +199,257 @@ global.mozIntl = {
-            totalTime -= sec * 1000;
-            // here we take tens of a msec only
-            ms = Math.round(totalTime / 10);
+          const format = durationFormats[type];
+          const duration = splitIntoTimeUnits(input, format.max, format.min);
+
+          if (duration.hasOwnProperty('millisecond')) {
+            // We round milliseconds to 2-digit
+            duration.millisecond =
+              Math.round(duration.millisecond / 10);
           }
 
           var string = fmt;
-          if (type === 'hms') {
-            string = string.replace('hh', numFormatter.format(hour));
-          }
-          string = string.replace('mm', numFormatter.format(min));
-          string = string.replace('ss', numFormatter.format(sec));
-          if (type === 'msS') {
-            string = string.replace('SS', numFormatter.format(ms));
-          }
+
+          format.tokens.forEach(([token, unit]) => {
+            string = string.replace(token,
+              numFormatter.format(duration[unit]));
+          });
           return string;
         }
       };
     });
+  },
+
+  /**
+   * RelativeTime formatter.
+   * Formats an integer with milliseconds into locale specific relative time 
+   * string.
+   *
+   * Currently accepted options:
+   *
+   * * style - long | short
+   *
+   * Defines whether the string will be long "1 minute ago" or short "1 min. 
+   * ago"
+   *
+   * * minUnit - (default) millisecond
+   * * maxUnit - (default) year
+   * * unit - second | minute | hour | day | week | month | year
+   *          (default) bestFit
+   *
+   * Example:
+   *
+   * const formatter = new Intl.RelativeTimeFormat(navigator.languages, {
+   *   unit: 'bestFit',
+   *   style: long'
+   * });
+   *
+   * var ms = Date.now() - 2 * 1000;
+   *
+   * formatter.format(ms); // "2 seconds ago"
+   *
+   * @param {Array} An array of languages
+   * @param {Array} Options object
+   * @returns An object with a formatter that returns Promises of strings
+   */
+  RelativeTimeFormat: function(locales, options) {
+    return {
+      /*
+       * ECMA 402 rev 3., 1.3.4, FormatRelativeTime
+       *
+       * Notes: This is a modified version of the function to use mozL10n
+       * and simplified to match current data set in data.properties
+       */
+      format: function(x) {
+        const ms = x - Date.now();
+        const units = computeTimeUnits(ms);
+
+        const unit = options.unit === 'bestFit' ?
+          getBestMatchUnit(units) : options.unit;
+
+        const v = units[unit];
+
+        // CLDR uses past || future
+        const tl = v < 0 ? '-ago' : '-until';
+        const style = options.style || 'long';
+
+        const entry = unit + 's' + tl + '-' + style;
+
+        return navigator.mozL10n.formatValue(entry, {
+          value: Math.abs(v)
+        });
+      },
+    };
+  },
+
+  _gaia: {
+    // This is an internal Firefox OS function, not part of the future standard
+    relativePart: function(milliseconds) {
```

Comment:
Eventually, yes, I agree. But currently I'm moving what's in l10n_date 1-1 and I'm not sure if those will be the destination function signatures, so I prefer to keep it very close and after 2.5 rethink what should happen with them.

I treat `_gaia` name space as "this is the old l10n_date.js".